site/app/controllers/pdf/PDFController.php

```diff
@@ -251,4 +304,155 @@ public function showGraderPDFEmbedded(string $gradeable_id) {
 
         $this->core->getOutput()->renderOutput(['PDF'], 'showPDFEmbedded', $gradeable_id, $id, $filename, $file_path, $anon_path, $anon_path, $annotation_jsons, false, $page_num, false, $is_peer_grader);
     }
+
+    #[Route(path: "/courses/{_semester}/{_course}/gradeable/{gradeable_id}/grading/img", methods: ["POST"])]
+    public function showGraderImageEmbedded(string $gradeable_id): void {
+        // User can be a team
+        $id = $_POST['user_id'] ?? null;
+        $filename = $_POST['filename'] ?? null;
+        // Explicit boolean parsing for is_anon
+        $is_anon = filter_var($_POST['is_anon'] ?? false, FILTER_VALIDATE_BOOL);
+        $filename = html_entity_decode($filename);
+        $file_path = urldecode($_POST['file_path']);
+        $real_path = $is_anon ? "" : $file_path;
+        $anon_path = $is_anon ? $file_path : "";
+
+        if ($is_anon) {
+            $id = $this->core->getQueries()->getSubmitterIdFromAnonId($id, $gradeable_id);
+            $real_path = $this->getPath($file_path, $id, true);
+            if (!file_exists($real_path)) {
+                $this->core->getOutput()->renderJsonFail('The image file could not be found');
+                return;
+            }
+        }
+        else {
+            $anon_path = $this->getPath($file_path, $gradeable_id, false);
```

Comment:
`getPath()` doesn't do any form of path sanitization when the third argument is false.  I believe that means that this code allows an arbitrary read if a carefully constructed `$_POST['file_path']` argument is provided.  It would be better to use `FileUtils::joinPaths()` in `getPath()` at a minimum, but it would be even better to just get rid of any sort of logic where we pass paths from the client side to the server side.  

Even if these security vulnerabilities are fixed now, passing paths around unnecessarily opens the door to developers unintentionally creating new security vulnerabilities in the future.

Reply:
The previous PDF annotator is unfortunately something we've had security issues with in the past, so it's best to not rely on it too much.  In general, I think passing the most minimal part needed to uniquely identify a file is the best way to approach that.  Is that just the filename, or a part of the filename?